fhir-server/src/main/antlr4/au/csiro/pathling/fhirpath/parser/generated/FhirPath.g4

```diff
@@ -145,16 +145,7 @@ STRING
         ;
 
 CODING
-        : CODING_WITHOUT_VERSION
-        | CODING_WITH_VERSION
-        ;
-
-fragment CODING_WITH_VERSION
-        : CODING_WITHOUT_VERSION '|' CODING_COMPONENT
-        ;
-
-fragment CODING_WITHOUT_VERSION
-        : CODING_COMPONENT '|' CODING_COMPONENT
+        : CODING_COMPONENT '|' CODING_COMPONENT ( '|' CODING_COMPONENT )*
```

Comment:
Should this be limited to at most five components?

Reply:
Strictly speaking yes. If it has more it will fail in the the Java code which actually enforces that.
I do not think that ANTLR regex support the syntax for specific number of repeats ( `X{n,m}`)  - I did tried the java syntax and it it did not work, and I could not find any alternative in the documentation.
Do you have any suggestions on how to rewrite this expression?
`CODING_COMPONENT '|' CODING_COMPONENT ( '|' CODING_COMPONENT )?  ( '|' CODING_COMPONENT )?  ( '|' CODING_COMPONENT )? ` ???

Reply:
I think your suggestion is the only way. I've made this change in a9e09ea54a0e21f8db4c48e01aa8e1bfecf0eef6.